tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/BaseLinker.scala

```diff
@@ -388,19 +389,34 @@ final class BaseLinker(semantics: Semantics, esLevel: ESLevel, considerPositions
         OptimizerHints.empty, targetMDef.hash)
   }
 
-  private def findInheritedMethodDef(classInfo: Analysis.ClassInfo,
-      methodName: String, getTree: TreeProvider,
+  private def findInheritedMethodDef(analysis: Analysis,
+      classInfo: Analysis.ClassInfo, methodName: String, getTree: TreeProvider,
       p: Analysis.MethodInfo => Boolean = _ => true): MethodDef = {
     @tailrec
     def loop(ancestorInfo: Analysis.ClassInfo): MethodDef = {
       assert(ancestorInfo != null,
           s"Could not find $methodName anywhere in ${classInfo.encodedName}")
 
       val inherited = ancestorInfo.methodInfos.get(methodName)
-      if (inherited.exists(p)) {
-        findMethodDef(ancestorInfo, methodName, getTree)
-      } else {
-        loop(ancestorInfo.superClass)
+      inherited.find(p) match {
+        case Some(m) =>
+          m.syntheticKind match {
+            case MethodSyntheticKind.None =>
+              findMethodDef(ancestorInfo, methodName, getTree)
+
+            case MethodSyntheticKind.DefaultBridge(targetInterface) =>
+              val targetInterfaceInfo = analysis.classInfos(targetInterface)
+              findMethodDef(targetInterfaceInfo, methodName, getTree)
```

Comment:
It happens all the time. That's because the reflective proxy is generated based on the target *name*, which causes a default bridge to be emitted for the reflective proxy. Consider this code:
```scala
trait Foo {
  def foo(): Int = 5
}

class Bar extends Foo

def test(bar: { def foo(): Int }): Unit =
  println(bar.foo())
```
The reflective call in `test` will try to find a reflective proxy for `foo__` in `Bar`. Since there is none yet, we look for a target. The target that is found is `Foo.foo`, *so the target name* is `foo__I`. Once the target name is found, we can generate the reflective proxy in `Bar`, which is:
```scala
class Bar extends Foo {
  def foo__(): any = this.foo__I()
}
```
Note that we remembered the *name* `foo__I`, but we "forgot" where we found it.

Now, that, in turns, is made reachable, and so we have a dynamic call to `Bar.foo__I`. That is not found either! So we look for a default method, and we find `Foo.foo__I`. We therefore generate a default bridge for that method in `Bar`, which becomes:
```scala
class Bar extends Foo {
  def foo__(): any = this.foo__I()

  def foo__I(): int = this.Foo::foo__I()
}
```
The default bridge `foo__I` knows that it's a bridge to `Foo.foo__I`. However, the reflective proxy only knows that it's a proxy to a method named `foo__I`, but we do not know where it is defined.

Therefore, when we come in the `BaseLinker`, we try and find the target of `Bar.foo__`, and we find, through inheritance, `Bar.foo__I`. That, in turn, is a default bridge, which we have to follow (through a direct link) to `Foo.foo__I`.